packages/headless/src/features/analytics/analytics-utils.ts

```diff
@@ -156,6 +160,11 @@ export type ProductListingAction<
   StateNeededByProductListingAnalyticsProvider
 >;
 
+export type ProductListingClickAction = PreparableAnalyticsAction<
+  {analyticsType: AnalyticsType.Click},
+  StateNeededBySearchAnalyticsProvider
+>;
```

Comment:
What is this for? I would expect `ProductListingAction<AnalyticsType.Click>` to be sufficient.

Also, should `StateNeededBySearchAnalyticsProvider` be `StateNeededByProductListingAnalyticsProvider`?

Reply:
Yes I made a mistake here and I forgot to use it. Thanks for noticing.

Reply:
So I removed it and used what you suggested instead.